packages/firestore/lib/src/query.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of firebase_firestore;
+
+/// Represents a query over the data at a particular location.
+class Query {
+  Query._({
+    @required Firestore firestore,
+    @required List<String> pathComponents,
```

Comment:
Is it ok for the required pathComponents to be empty or null? Maybe add an assert just like it was added for firestore?

Reply:
Done